uPortal-security/uPortal-security-mvc/src/main/java/org/apereo/portal/url/MaxInactiveFilter.java

```diff
@@ -0,0 +1,127 @@
+/*
+ Licensed to Apereo under one or more contributor license agreements. See the NOTICE file
+ distributed with this work for additional information regarding copyright ownership. Apereo
+ licenses this file to you under the Apache License, Version 2.0 (the "License"); you may not use
+ this file except in compliance with the License. You may obtain a copy of the License at the
+ following location:
+
+ <p>http://www.apache.org/licenses/LICENSE-2.0
+
+ <p>Unless required by applicable law or agreed to in writing, software distributed under the
+ License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ express or implied. See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+package org.apereo.portal.url;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpSession;
+import lombok.extern.slf4j.Slf4j;
+import org.apereo.portal.security.IPerson;
+import org.apereo.portal.security.IPersonManager;
+import org.apereo.portal.security.ISecurityContext;
+import org.springframework.beans.factory.annotation.Autowired;
+
+/**
+ * This is a refactor to correct the application of the session timeout (aka MaxInactive time) to
+ * make sure it is applied consistently. The {@code HandlerInterceptorAdapter} version was not
+ * applying correctly to JSON and other corner cases.
+ *
+ * @since 5.7.1
+ */
+@Slf4j
+public class MaxInactiveFilter implements Filter {
+
+    private static final ZoneId tz = ZoneId.systemDefault();
+
+    // IPerson attribute key to flag if this value has already been set
+    public static final String SESSION_MAX_INACTIVE_SET_ATTR = "MAX_INACTIVE_SET";
+
+    @Autowired private IPersonManager personManager;
+
+    @Autowired private IMaxInactiveStrategy maxInactiveStrategy;
+
+    public static final int REFRESH_MINUTES = 5;
+
+    @Override
+    public void init(FilterConfig filterConfig) {}
+
+    @Override
+    public void destroy() {}
+
+    @Override
+    public void doFilter(ServletRequest req, ServletResponse resp, FilterChain filterChain)
+            throws IOException, ServletException {
+        log.info("Entering MaxInactiveFilter");
+        final HttpServletRequest request = (HttpServletRequest) req;
+        checkMaxInactive(request);
+
+        log.info("Continuing on to other filters from MaxInactiveFilter");
```

Comment:
Probably want this as `log.debug` as well.

Reply:
:+1: